include/geometric_shapes/bodies.h

```diff
@@ -225,7 +225,7 @@ class Body
      Sometimes multiple attempts need to be generated.
      The function terminates with failure (returns false) after \e max_attempts attempts.
      If the call is successful (returns true) the point is written to \e result */
-  virtual bool samplePointInside(random_numbers::RandomNumberGenerator& rng, unsigned int max_attempts,
+  virtual bool samplePointInside(std::function<double(double, double)> random_value, unsigned int max_attempts,
```

Comment:
Readability is reduced here: What's the meaning of the arguments and return value of the std::function?
If this approach is to be followed, there should be a type definition for this function type explaining the args.

Reply:
Good point, I've updated the comment